### PR TITLE
feat: update explorer icon and add paste to custom destination

### PR DESCRIPTION
**Pull request by RyukTheCoder**
# Summary

- improve explorer icon and tooltip in swap details page
- add paste button to custom destination for swap
- improve swapper display name (showing title instead of swapperGroup)

Fixes # RF-1434

# How did you test this change?

Checked every item in widget.

# Checklist:

- [x] I have performed a self-review of my code
- [x] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWallets.styles.ts [hunk 1]
```diff
@@ -118,7 +118,14 @@ export const ConfirmButton = styled('div', {
...
   backgroundColor: '$neutral100',
-  padding: '$0 $15 $15 $15',
+  paddingLeft: '$15',
+  variants: {
+    show: {
+      false: {
+        visibility: 'hidden',
+      },
+    },
+  },
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/ConfirmWalletsModal/ConfirmWallets.styles.ts:128]: The padding of the text field doesn't align with the UI design. Additionally, there's a shift in the position of the paste icon when opening and closing the collapsible.

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx [hunk 2]
```diff
@@ -438,9 +459,22 @@ export function ConfirmWalletsModal(props: PropTypes) {
...
                         <StyledTextField
+                          show={showCustomDestination}
+                          ref={customDestinationInputRef}
+                          style={{ padding: 0 }}
                           autoFocus
                           placeholder={i18n.t('Your destination address')}
                           value={customDestination || ''}
+                          suffix={
+                            !isFirefox &&
+                            (!isMobile || !customDestination) && (
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:470]: Is this condition accurate?
`(!isMobile || !customDestination)`

File: widget/embedded/src/components/CustomCollapsible/CustomCollapsible.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ import {
 export function CustomCollapsible(props: PropsWithChildren<PropTypes>) {
   const {
     open,
+    outlined = true,
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/CustomCollapsible/CustomCollapsible.tsx:15]: Do we really need this property? Currently, when the custom destination is open, there's no border around it. Please review the next branch. I believe the UI design is incorrect.

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx [hunk 4]
```diff
@@ -230,6 +269,10 @@ export function ConfirmWalletsModal(props: PropTypes) {
     );
   }, [connectedWallets.length]);
 
+  useEffect(() => {
+    setIsFirefox(navigator?.userAgent.includes('Firefox'));
+  }, []);
+
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:275]: Is the "isFirefox" state and this "useEffect" necessary? Can we do without them?
`const isFirefox = navigator?.userAgent.includes('Firefox')`

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx [hunk 5]
```diff
@@ -138,6 +146,37 @@ export function ConfirmWalletsModal(props: PropTypes) {
...
   };
 
+  const handlePasteCustomDestination = async (
+    event: MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+    if (navigator.clipboard !== undefined) {
+      const pastedText = await navigator.clipboard.readText();
+      setCustomDestination(pastedText);
+      customDestinationInputRef?.current?.focus();
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:156]: Currently, if there is no text in the clipboard, we show the paste icon. Can we modify it to display the paste icon only if there is data in the clipboard?

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx (hunk 2 above)
Comment (RyukTheCoder) [reply widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:470]: Yes, in mobile view we should show the button only if the user has not entered any value.

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx (hunk 4 above)
Comment (RyukTheCoder) [reply widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:275]: You are right. Fixed.

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx (hunk 5 above)
Comment (RyukTheCoder) [reply widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:156]: I don't think that this is necessary. I have not seen anything similar to what you say.

File: widget/embedded/src/components/SwapDetails/SwapDetails.tsx [hunk 6]
```diff
@@ -322,7 +323,11 @@ export function SwapDetails(props: SwapDetailsProps) {
               <StyledLink
                 target="_blank"
                 href={`${SCANNER_BASE_URL}/swap/${requestId}`}>
-                <LinkIcon size={16} />
+                <Tooltip
+                  content={i18n.t('View on Rango Explorer')}
+                  side="bottom">
+                  <RangoExplorerIcon size={16} />
+                </Tooltip>
```
[the remark this answers is not in the document]
Comment (RyukTheCoder) [reply widget/embedded/src/components/SwapDetails/SwapDetails.tsx:330]: Tnx, Fixed.

File: widget/embedded/src/components/CustomCollapsible/CustomCollapsible.tsx (hunk 3 above)
Comment (RyukTheCoder) [reply widget/embedded/src/components/CustomCollapsible/CustomCollapsible.tsx:15]: Removed

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWallets.styles.ts (hunk 1 above)
Comment (RyukTheCoder) [reply widget/embedded/src/components/ConfirmWalletsModal/ConfirmWallets.styles.ts:128]: Fixed the layout shift. please check if you are happy with the paddings.

**Review by Ikari-Shinji-re — APPROVED** [review]

**Comment by samobasquiat** [comment]
It seems that the input placeholder should be changed based on the destination chain.
image

**Comment by samobasquiat** [comment]
When we open the input box the title color has changed. 
image

**Comment by samobasquiat** [comment]
I noticed a minor shift when I closed the box. @Ikari-Shinji-re commented, but I don't know if it relates to that.
image

**Comment by samobasquiat** [comment]
I think in the dark mode the color of paste icon is different with Figma.

**Comment by samobasquiat** [comment]
After pasting I don't see the Icon in Figma. I don't know it's a bug or not.
image

**Comment by samobasquiat** [comment]
We have a bug that is not related to this PR but when we clean the input text the box is closed but the destination wallet is not selected again.

**Review by samobasquiat — CHANGES_REQUESTED** [review]

File: widget/embedded/src/components/SwapDetails/SwapDetails.tsx [hunk 7]
```diff
@@ -322,7 +323,12 @@ export function SwapDetails(props: SwapDetailsProps) {
               <StyledLink
                 target="_blank"
                 href={`${SCANNER_BASE_URL}/swap/${requestId}`}>
-                <LinkIcon size={16} />
+                <Tooltip
+                  container={getContainer()}
+                  content={i18n.t('View on Rango Explorer')}
+                  side="bottom">
+                  <RangoExplorerIcon size={16} />
```
[the remark this answers is not in the document]
Comment (RyukTheCoder) [reply widget/embedded/src/components/SwapDetails/SwapDetails.tsx:330]: Fixed

**Comment by RyukTheCoder** [comment]
> I noticed a minor shift when I closed the box. @Ikari-Shinji-re commented, but I don't know if it relates to that. [Screencast.webm](https://github.com/rango-exchange/rango-client/assets/123297941/e5e41c8d-b5ab-4af4-ac16-f138197dabb2)

It is fixed in the last deployment.

**Comment by RyukTheCoder** [comment]
> After pasting I don't see the Icon in Figma. I don't know it's a bug or not. image

In an offline conversation with the design team, discussed that on mobile view we should only show paste icon if the address is not typed. So in desktop I'm showing the paste button even if user has typed something.

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx [hunk 8]
```diff
@@ -477,8 +478,13 @@ export function ConfirmWalletsModal(props: PropTypes) {
...
                           suffix={
-                            !isFirefox &&
-                            (!isMobile || !customDestination) && (
+                            customDestination ? (
+                              <IconButton
+                                onClick={handleClearCustomDestination}
+                                variant="ghost">
+                                <ClearIcon size={16} />
+                              </IconButton>
+                            ) : !!isFirefox ? null : (
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:487]: Reading nested ternary can be challenging

**Comment by Ikari-Shinji-re** [comment]
I believe these icons should be vertically aligned together. Please check the UI design for confirmation:

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx [hunk 9]
```diff
@@ -420,25 +441,57 @@ export function ConfirmWalletsModal(props: PropTypes) {
...
+                              ),
+                            }
+                          )}
                           value={customDestination || ''}
+                          suffix={
+                            customDestination ? (
+                              <IconButton
+                                onClick={handleClearCustomDestination}
+                                variant="ghost">
+                                <ClearIcon size={16} />
```
Comment (RanGojo) [inline widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:485]: <ClearIcon size={12} />
```suggestion
                                <ClearIcon size={16} />
```

Comment (RanGojo) [inline widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:485]:
```suggestion
                                <ClearIcon size={12} />
```

File: widget/ui/src/icons/index.ts [hunk 10]
```diff
@@ -7,6 +7,7 @@ export { default as ChevronDownIcon } from './ChevronDown';
 export { default as ChevronLeftIcon } from './ChevronLeft';
 export { default as ChevronRightIcon } from './ChevronRight';
 export { default as ChevronUpIcon } from './ChevronUp';
+export { default as ClearIcon } from './Clear';
```
Comment (RanGojo) [inline widget/ui/src/icons/index.ts:10]: I think this is not required. you could use close icon.
also make sure that both clear and copy icon have same hover effect/color in confirm wallet page.

File: widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx (hunk 8 above)
Comment (RyukTheCoder) [reply widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx:487]: Tnx, Fixed.

**Pull request merged**